.changeset/warm-glasses-lie.md

```diff
@@ -0,0 +1,9 @@
+---
+'@backstage/plugin-scaffolder-backend': minor
+---
+
+-`projectid` is now passed through the query parameter `project` in the `repoUrl`. It still allows people to not use the `projectid` and use the `repoUrl` with the `owner` and `repo` query parameters instead. This makes it easier to publish to repositories instead of writing the full path to the project.
+
+- Changed the `parseUrl` function so it uses a `switch-case` statement which makes the function more readable than having several `if-else` statements.
+- Created an auxiliary function called `checkRequiredParams`. The function checks if the required parameters (given as arguments) are present in the URL.
+- The function `parseUrl` uses `checkRequiredParams` to verify that all the required query parameters are present in the repo's URL. This replaces the `if-else` mess with a `switch` statement for better readability.
```

Comment:
Thanks for the additional context here but it does not have to go into the changelog for this package as it's internal changes. They're a much better fit in  the PR description :) 

```suggestion
```

Reply:
Hello @jhaals ! Thank you for reviewing!
As to your main question, I have made some tests locally and the change I made work both when I pass `project` as a query parameter and when I pass `owner` and `repo` together. Do you need proof/screenshots showing how this works?

Reply:
Made the requested change of removing the `.changeset` file